apps/api/src/domain/models/media-item/commands/create-media-item.command.integration.spec.ts

```diff
@@ -0,0 +1,335 @@
+                                {
+                                    contributorId: '33',
+                                    role: ['screenplay', 'editor'],
+                                },
+                            ],
+                        }),
+                    initialState: emptyInitialState,
+                    checkError: (error) => assertCommandPayloadTypeError(error, 'contributions'),
+                });
+            });
+        });
+
+        describe('when the MIME type is not one of the registered types', () => {
+            it('should fail', async () => {
+                await assertCreateCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: (id: AggregateId) =>
+                        buildInvalidFSA(id, {
+                            mimeType: 'bogus/mp12',
+                        }),
+                    initialState: emptyInitialState,
+                    checkError: (error) => assertCommandPayloadTypeError(error, 'mimeType'),
+                });
+            });
+        });
+
+        describe('when RawData is a string', () => {
+            it('should fail', async () => {
+                await assertCreateCommandError(assertionHelperDependencies, {
+                    buildCommandFSA: (id: AggregateId) =>
+                        buildInvalidFSA(id, {
+                            rawData: 'I should really be an object!',
+                        }),
+                    initialState: emptyInitialState,
+                    checkError: (error) => assertCommandPayloadTypeError(error, 'rawData'),
+                });
+            });
+        });
+    });
+
+    describe('when the external state is invalid', () => {
+        it('should fail', async () => {
+            await assertCreateCommandError(assertionHelperDependencies, {
+                buildCommandFSA: (_: AggregateId) => buildValidCommandFSA(existingMediaItemId),
+                initialState: buildInMemorySnapshot({
+                    resources: {
+                        mediaItem: [
+                            getValidResourceInstanceForTest(ResourceType.mediaItem).clone({
+                                id: existingMediaItemId,
+                            }),
+                        ],
+                    },
+                }),
+
+                checkError: (error) => {
+                    expect(error).toBeInstanceOf(CommandExecutionError);
+
+                    expect(error.innerErrors.length).toBe(1);
+
+                    const innerError = error.innerErrors[0];
+
+                    expect(innerError).toEqual(
+                        new ResourceIdAlreadyInUseError({
+                            id: existingMediaItemId,
+                            resourceType: ResourceType.mediaItem,
+                        })
+                    );
+                },
+            });
+        });
+    });
+
+    describe('when an invariant validation rule is not satisfied by the new media item', () => {
```

Comment:
We should make a `command test coverage checklist` and use this each time we write a new command test suite.